packages/button/src/Button.ts

```diff
@@ -136,4 +177,15 @@ export class Button extends SizedMixin(ButtonBase) {
             this.setAttribute('variant', this.variant);
         }
     }
+
+    protected override renderButton(): TemplateResult {
+        return html`
+            ${this.buttonContent}
+            <sp-progress-circle
+                indeterminate
+                static="white"
+                size="s"
```

Comment:
We likely want to match the t-shirt size approach to that of #3110 as well.

Reply:
Yeah i had a play with this, was 'tricky', one step down in size was not always the rule, and the sizes didn't seem to align well.

Reply:
Did some more testing here and there's no good sizes here that work. `s` size progress-circle on `s` sized button is too big, `s` on `m` button is about right, but `m` on `l` button is too big.

Not sure what to do here, we could apply `scale` on the transform for the progress-circle to tweak it at each t-shirt size perhaps? But it'll likely be a custom value per button size.